src/map/utils/battleutils.cpp

```diff
@@ -1921,35 +1921,20 @@ namespace battleutils
         CItemWeapon* PWeapon = GetEntityWeapon(PDefender, SLOT_MAIN);
         if (((PDefender->objtype == TYPE_PC && PWeapon != nullptr && PWeapon->getID() != 0 && PWeapon->getID() != 65535 && PWeapon->getSkillType() != SKILL_HAND_TO_HAND) ||
              (PDefender->objtype == TYPE_MOB && PDefender->m_EcoSystem == ECOSYSTEM::BEASTMAN && PDefender->GetMJob() != JOB_MNK && PDefender->isInDynamis())) &&
-            PDefender->PAI->IsEngaged())
+              PDefender->PAI->IsEngaged())
         {
-            // http://wiki.ffxiclopedia.org/wiki/Talk:Parrying_Skill
-            // {(Parry Skill x .125) + ([Player Agi - Enemy Dex] x .125)} x Diff
+            float  defender_parry_skill = (float)(PDefender->GetSkill(SKILL_PARRY) + PDefender->getMod(Mod::PARRY) + PWeapon->getILvlParry());
+            uint16 attackSkill          = PAttacker->GetSkill((SKILLTYPE)(weapon ? weapon->getSkillType(): 0 ));
```

Comment:
you'll wanna declare `weapon` as `CItemWeapon* weapon = GetEntityWeapon(PAttacker, SLOT_MAIN)`, but i don't know enough about the codebase to tell you what edge-cases to look out for (i see above they're checking for nullptr and ids, idk what those cases are supposed to indicate)

Reply:
This is largely a rename of already existing code (before the this reformulation), will go through and try and discern the reason for the checks above when I make the change.

Reply:
Adding the weapon declaration in a commit and squashing, still somewhat unsure about the nullptr and where to include it outside the opening segment as it pertains to ensuring that the defender has a weapon equipped.